### ISSUE TITLE
Download in background

### ISSUE THREAD

**Issue by danielmmmm** [n76/Local-GSM-Backend#7]
Hi!
I just started using UnifiedNlp and your backend and I am trying to download databases. The download of OpenCellID finishes immediately with a download length of -1 ... don't know what's wrong here. Downloading the Mozilla database works, but every time Local-GSM-Backend loses focus (e.g. when I receive an SMS that is displayed as a pop-up), I have to start the download again, from the very beginning.

**Comment by n76** [comment]
Definitely a bug on having to start over again when losing focus. I will look into that.

With respect to OpenCellID, do you have an API code set? Also, they only allow one download per day so if you failed at some point (maybe a SMS message received. . .) it could give you a -1 length on subsequent attempts.

**Comment by danielmmmm** [comment]
Great, thanks!

I will try OpenCellID again tomorrow or the day after. I have an ID and I was able to download the database on my PC. I could probably copy it to my phone and use it in your backend.

**Comment by n76** [comment]
is your "PC" Unix based or has software that can run Unix scripts? If so, then you might try running https://github.com/n76/lacells-creator/blob/master/gen_lacells_merged to build the database. The advantage being that it won't be interrupted by SMS and if you already have data downloaded from OpenCellID you can comment out that section of the script.

**Comment by danielmmmm** [comment]
I still get the same error from OpenCellID. Also, when both data sources are selected and OpenCellID fails, it won't attempt to download the Mozilla database.
I managed to build a database on my PC (Windows with Cygwin) and push it to my phone. Although I have no idea if I did everything correctly because I can't get UnifiedNlp to work (I have Google Play services on my phone and am trying to get UnifiedNlp to work with Rawi01's Xposed module ... with no luck so far).

Another thing: From your script I get that setting MCCs is a destructive process. Would it be feasible to save the original (unfiltered) databases as well? Of course they would have to be compressed. In any case, is there a possibility to save databases on the external SD card? Your module seems to use sdcard0 only.
